### PR TITLE
RNDA-272. Implement synchronization operations in the driver pt. 2

### PR DESCRIPTION
**Pull request by kbiczk-mobica**
Implement synchronization during network TX/RX operations.
Add ATHN_LOCK ATHN_UNLOCK ATHN_ASSERT macros instead of the standard function call.

### CODE REVIEW

**Comment by Mikolaj-Filar-Mobica** [comment]
Test

File: sys/dev/athn/if_athn_usb.c [hunk 1]
```diff
@@ -578,13 +578,13 @@ athn_usb_attachhook(device_t self)
 	}
 
 	// TODO MichalP: this can be used as a starting point for echo command or firmware command
-//	mtx_lock(&usc->sc_sc.sc_mtx);
+//	ATHN_LOCK(usc);
```
Comment (Mikolaj-Filar-Mobica) [inline sys/dev/athn/if_athn_usb.c]: Shouldn't it be `ATHN_LOCK(&usc->sc_sc);`.?

**Comment by Mikolaj-Filar-Mobica** [comment]
I think these changes are in conflict with Norbert's changes in branch RDNA-274. Suggest talking with Norbert before merging these.
**Update**: Norbert agrees to merge that

File: sys/dev/athn/if_athn_usb.c (hunk 1 above)
Comment (kbiczk-mobica) [reply sys/dev/athn/if_athn_usb.c]: Indeed, you have the right. Change is on the way :)

**Review by Mikolaj-Filar-Mobica — APPROVED** [review]

**Pull request merged**